operate/webapp/src/main/java/io/camunda/operate/webapp/api/v1/dao/opensearch/OpensearchSearchableDao.java

```diff
@@ -86,17 +92,57 @@ protected void buildSorting(Query<T> query, String uniqueSortKey, SearchRequest.
     request.sort(queryDSLWrapper.sortOptions(uniqueSortKey, SortOrder.Asc));
   }
 
-  protected void buildPaging(Query<T> query, SearchRequest.Builder request) {
+  protected void buildPaging(final Query<T> query, final SearchRequest.Builder request) {
     final Object[] searchAfter = query.getSearchAfter();
     if (searchAfter != null) {
       request.searchAfter(CollectionUtil.toSafeListOfStrings(searchAfter));
     }
     request.size(query.getSize());
   }
 
-  protected abstract void buildFiltering(Query<T> query, SearchRequest.Builder request);
+  /**
+   * Builds a tenant-aware filter query based on the query terms provided by {@link
+   * #collectFilterQueryTerms(Object)} and {@link #collectRequiredFilterQueryTerms()}
+   */
+  @VisibleForTesting
+  protected void buildFiltering(final Query<T> query, final SearchRequest.Builder request) {
+    final var queryTerms =
+        Stream.concat(
+                collectRequiredFilterQueryTerms(),
+                Optional.ofNullable(query.getFilter())
+                    .map(this::collectFilterQueryTerms)
+                    .orElseGet(Stream::empty))
+            .filter(Objects::nonNull)
+            .toList();
+
+    if (!queryTerms.isEmpty()) {
+      request.query(queryDSLWrapper.withTenantCheck(queryDSLWrapper.and(queryTerms)));
+    }
+  }
+
+  /**
+   * Collect required query terms that must always be present.
+   *
+   * <p>You can use {@link #collectFilterQueryTerms(Object)} instead to build query terms based on
+   * an existing filter object. The two may be combined if you have query terms that always need to
+   * be present as well as query terms based on a filter object that may or may not exist.
```

Comment:
🔧 We might document here that if no query terms are provided, then the tenant check is added.